panda/crypto/panda.go

```diff
@@ -178,77 +167,80 @@ func (kx *KeyExchange) SetSharedRandom(srv []byte) {
 }
 
 func (kx *KeyExchange) Run() {
```

Comment:
maybe this function should be able to signal to the caller that it hit an error?

Reply:
Yes, perhaps but this isn't the topic for this PR

Reply:
I think that those fixes ended up here while I needed to fix races in panda_test.go I also noted that the panda client deadlocks at shutdown. Errors are still returned to the caller via panda update channel, but I agree it's a clunky API, but if I make changes to this now/here I have to update many other things as well.